src/odbc_connection.cpp

```diff
@@ -2844,108 +2913,399 @@ fetch_and_store
           // Iterate over each column, putting the data in the row object
           for (int column_index = 0; column_index < data->column_count; column_index++)
           {
-            if (data->bound_columns[column_index].length_or_indicator_array[row_index] == SQL_NULL_DATA) {
-              row[column_index].size = SQL_NULL_DATA;
-            } else {
-              switch (data->columns[column_index]->bind_type) {
-
-                case SQL_C_DOUBLE:
-                  row[column_index].double_data =
-                    ((SQLDOUBLE *)(data->bound_columns[column_index].buffer))[row_index];
-                  break;
-
-                case SQL_C_UTINYINT:
-                  row[column_index].tinyint_data =
-                    ((SQLCHAR *)(data->bound_columns[column_index].buffer))[row_index];
-                  break;
-
-                case SQL_C_SSHORT:
-                case SQL_C_SHORT:
-                  row[column_index].smallint_data =
-                    ((SQLSMALLINT *)(data->bound_columns[column_index].buffer))[row_index];
-                  break;
-
-                case SQL_C_USHORT:
-                  row[column_index].usmallint_data =
-                    ((SQLUSMALLINT *)(data->bound_columns[column_index].buffer))[row_index];
-                  break;
-
-                case SQL_C_SLONG:
-                  row[column_index].integer_data =
-                    ((SQLINTEGER *)(data->bound_columns[column_index].buffer))[row_index];
-                  break;
-
-                case SQL_C_UBIGINT:
-                  row[column_index].ubigint_data =
-                    ((SQLBIGINT *)(data->bound_columns[column_index].buffer))[row_index];
-                  break;
-
-                case SQL_C_BINARY:
+            // The column contained SQL_(W)LONG* data, so we didn't call
+            // SQLBindCol, and therefore there is no data to move from a buffer.
+            // Instead, call SQLGetData, and adjust buffer size accordingly
+            if (data->columns[column_index]->is_long_data) {
+
+              // Get the first chunk of data
+              data->sqlReturnCode =
+              SQLGetData
+              (
+                data->hSTMT,
+                column_index + 1,
+                data->columns[column_index]->bind_type,
+                data->long_data_buffer,
+                data->long_data_buffer_size,
+                &data->long_data_string_length_or_indicator
```

Comment:
Is there a reason this indicator needs to be on the data object? Seems like a local would do, since it's going to get copied in to the row size.